engine/src/main/java/org/terasology/engine/rendering/primitives/BlockMeshShapeGenerator.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 The Terasology Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+package org.terasology.engine.rendering.primitives;
+
+import org.terasology.engine.rendering.assets.mesh.Mesh;
+import org.terasology.engine.rendering.assets.mesh.StandardMeshData;
+import org.terasology.engine.utilities.Assets;
+import org.terasology.engine.world.block.Block;
+import org.terasology.engine.world.block.BlockPart;
+import org.terasology.engine.world.block.shapes.BlockMeshPart;
+import org.terasology.gestalt.assets.ResourceUrn;
+import org.terasology.nui.Color;
+
+public abstract class BlockMeshShapeGenerator implements BlockMeshGenerator {
+    protected Mesh mesh = null;
+
+    public abstract Block getBlock();
+
+    public abstract ResourceUrn baseUrn();
+
+    @Override
+    public Mesh getStandaloneMesh() {
+        if (mesh == null || mesh.isDisposed()) {
+            Block block = getBlock();
+            StandardMeshData meshData = new StandardMeshData();
+            int nextIndex = 0;
+            for (BlockPart dir : BlockPart.allParts()) {
+                BlockMeshPart part = block.getPrimaryAppearance().getPart(dir);
+                if (part != null) {
+                    for (int i = 0; i < part.size(); i++) {
+                        meshData.position.put(part.getVertex(i));
+                        meshData.color0.put(Color.white);
+                        meshData.normal.put(part.getNormal(i));
+                        meshData.uv0.put(part.getTexCoord(i));
+                    }
```

Comment:
The [`meshData.light0` change](https://github.com/MovingBlocks/Terasology/blob/c725b454517bcaec881c343d5045eb76d722868b/engine/src/main/java/org/terasology/engine/rendering/primitives/Tessellator.java#L122-L124) of the original Tesselator `addMeshPart()` method is missing, is this intended?

Reply:
light0 and light1 just change the color of certain vertices its pretty specific to BlockMeshShape.

Reply:
umm, do we use light0 seems like its a fixed value. if so we should just remove it since it blows up the size of the mesh data.